File: LeetCode/Reverse Linked List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
    var prev *ListNode = nil
    var next *ListNode = nil
    
    curr := head
    
    if (curr == nil){
        return curr
    }
    
    for curr.Next != nil {
        next = curr.Next
        curr.Next = prev
        prev = curr
        curr = next
    }
    
    curr.Next = prev
    
    return curr
}

File: LeetCode/Reverse Linked List.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function reverseList(head: ListNode | null): ListNode | null {
  let prev = null,
    next = null,
    curr = head;

  if (!curr) return curr;

  while (curr.next) {
    next = curr.next;
    curr.next = prev;
    prev = curr;
    curr = next;
  }

  curr.next = prev;

  return curr;
}
